src/vidpf.rs

```diff
@@ -76,6 +82,11 @@ impl<W: VidpfValue, const NONCE_SIZE: usize> Vidpf<W, NONCE_SIZE> {
         Self { weight_parameter }
     }
 
+    /// Public accessor method for the weight_parameter field
+    pub(crate) fn weight_parameter(&self) -> &W::ValueParameter {
+        &self.weight_parameter
+    }
```

Comment:
Similar point as elsewhere: it may be clearer to make `weight_parameter` `pub(crate)` and drop this accessor.